expirer/src/main/scala/com/gu/media/expirer/ExpirerLambda.scala

```diff
@@ -0,0 +1,71 @@
+package com.gu.media.expirer
+
+import java.time.Instant
+
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import com.gu.media.CapiPreviewAccess
+import com.gu.media.lambda.LambdaBase
+import com.gu.media.logging.Logging
+import com.gu.media.youtube.{YouTubeAccess, YouTubeVideos}
+import play.api.libs.json.{JsArray, JsValue}
+
+import scala.annotation.tailrec
+import scala.util.control.NonFatal
+
+class ExpirerLambda extends RequestHandler[Unit, Unit]
+  with LambdaBase
+  with Logging
+  with CapiPreviewAccess
+  with YouTubeAccess
+  with YouTubeVideos {
+
+  def expireInParallel = true // disabled in unit tests
+
+  override def handleRequest(input: Unit, context: Context): Unit = {
+    val epochMillis = Instant.now().toEpochMilli
+    val assets = getExpiredAssets(100, 1, epochMillis, Set.empty)
+
+    val toExpire = if(expireInParallel) { assets.par } else { assets }
+
+    toExpire.foreach { video =>
+      try {
+        setStatusToPrivate(video)
+      } catch {
+        case NonFatal(err) =>
+          log.error(s"Unable to expire $video", err)
+      }
+    }
+  }
+
+  @tailrec
+  private def getExpiredAssets(pageSize: Int, page: Int, now: Long, before: Set[String]): Set[String] = {
```

Comment:
It accumulates a set of all the video ids referenced by expired atoms. Perhaps `getExpiredVideos` is clearer?